<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Importing microbiome data | Microbiome data science with R/Bioconductor</title>
  <meta name="description" content="Course material" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Importing microbiome data | Microbiome data science with R/Bioconductor" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Course material" />
  <meta name="github-repo" content="microbiome/course_2022_oulu" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Importing microbiome data | Microbiome data science with R/Bioconductor" />
  
  <meta name="twitter:description" content="Course material" />
  



<meta name="date" content="2022-07-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="reproducible-reporting-with-rmarkdown.html"/>
<link rel="next" href="microbiome-data-exploration.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Microbiome data science with R/Bioconductor</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#contents-and-learning-goals"><i class="fa fa-check"></i><b>1.1</b> Contents and learning goals</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#schedule-and-organizers"><i class="fa fa-check"></i><b>1.2</b> Schedule and organizers</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#acknowledgments"><i class="fa fa-check"></i><b>1.3</b> Acknowledgments</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="code-of-conduct.html"><a href="code-of-conduct.html"><i class="fa fa-check"></i><b>2</b> Code of Conduct</a></li>
<li class="chapter" data-level="3" data-path="start.html"><a href="start.html"><i class="fa fa-check"></i><b>3</b> Getting started</a>
<ul>
<li class="chapter" data-level="3.1" data-path="start.html"><a href="start.html#checklist-before-the-course"><i class="fa fa-check"></i><b>3.1</b> Checklist (before the course)</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="start.html"><a href="start.html#computer-setup-and-installations"><i class="fa fa-check"></i><b>3.1.1</b> Computer setup and installations</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="start.html"><a href="start.html#support-and-resources"><i class="fa fa-check"></i><b>3.2</b> Support and resources</a></li>
<li class="chapter" data-level="3.3" data-path="start.html"><a href="start.html#packages"><i class="fa fa-check"></i><b>3.3</b> Installing and loading the required R packages</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="reproducible-reporting-with-rmarkdown.html"><a href="reproducible-reporting-with-rmarkdown.html"><i class="fa fa-check"></i><b>4</b> Reproducible reporting with Rmarkdown</a></li>
<li class="chapter" data-level="5" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html"><i class="fa fa-check"></i><b>5</b> Importing microbiome data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#data-access"><i class="fa fa-check"></i><b>5.1</b> Data access</a></li>
<li class="chapter" data-level="5.2" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#importing-microbiome-data-in-r"><i class="fa fa-check"></i><b>5.2</b> Importing microbiome data in R</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#example-solution"><i class="fa fa-check"></i><b>5.2.1</b> Example solution</a></li>
<li class="chapter" data-level="5.2.2" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#loading-readily-processed-data"><i class="fa fa-check"></i><b>5.2.2</b> Loading readily processed data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html"><i class="fa fa-check"></i><b>6</b> Microbiome data exploration</a>
<ul>
<li class="chapter" data-level="6.1" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#data-structure"><i class="fa fa-check"></i><b>6.1</b> Data structure</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#transformations"><i class="fa fa-check"></i><b>6.1.1</b> Transformations</a></li>
<li class="chapter" data-level="6.1.2" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#aggregation"><i class="fa fa-check"></i><b>6.1.2</b> Aggregation</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#visualization"><i class="fa fa-check"></i><b>6.2</b> Visualization</a></li>
<li class="chapter" data-level="6.3" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#exercises-optional"><i class="fa fa-check"></i><b>6.3</b> Exercises (optional)</a></li>
<li class="chapter" data-level="6.4" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#exploring-data-example-solutions"><i class="fa fa-check"></i><b>6.4</b> Exploring data: example solutions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="alpha-diversity-demo.html"><a href="alpha-diversity-demo.html"><i class="fa fa-check"></i><b>7</b> Alpha diversity demo</a>
<ul>
<li class="chapter" data-level="7.1" data-path="alpha-diversity-demo.html"><a href="alpha-diversity-demo.html#alpha-diversity-estimation"><i class="fa fa-check"></i><b>7.1</b> Alpha diversity estimation</a></li>
<li class="chapter" data-level="7.2" data-path="alpha-diversity-demo.html"><a href="alpha-diversity-demo.html#visualizing-alpha-diversity"><i class="fa fa-check"></i><b>7.2</b> Visualizing alpha diversity</a></li>
<li class="chapter" data-level="7.3" data-path="alpha-diversity-demo.html"><a href="alpha-diversity-demo.html#comparing-alpha-diversity"><i class="fa fa-check"></i><b>7.3</b> Comparing alpha diversity</a></li>
<li class="chapter" data-level="7.4" data-path="alpha-diversity-demo.html"><a href="alpha-diversity-demo.html#exercises"><i class="fa fa-check"></i><b>7.4</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity-demo.html"><a href="beta-diversity-demo.html"><i class="fa fa-check"></i><b>8</b> Beta diversity demo</a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity-demo.html"><a href="beta-diversity-demo.html#visualizations"><i class="fa fa-check"></i><b>8.1</b> Visualizations</a></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity-demo.html"><a href="beta-diversity-demo.html#hypothesis-testing"><i class="fa fa-check"></i><b>8.2</b> Hypothesis testing</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity-demo.html"><a href="beta-diversity-demo.html#testing-the-differences-in-dispersion"><i class="fa fa-check"></i><b>8.2.1</b> Testing the differences in dispersion</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity-demo.html"><a href="beta-diversity-demo.html#exercises-1"><i class="fa fa-check"></i><b>8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="material.html"><a href="material.html"><i class="fa fa-check"></i><b>9</b> Study material</a>
<ul>
<li class="chapter" data-level="9.1" data-path="material.html"><a href="material.html#online-tutorial"><i class="fa fa-check"></i><b>9.1</b> Online tutorial</a></li>
<li class="chapter" data-level="9.2" data-path="material.html"><a href="material.html#lecture-slides"><i class="fa fa-check"></i><b>9.2</b> Lecture slides</a></li>
<li class="chapter" data-level="9.3" data-path="material.html"><a href="material.html#tasks"><i class="fa fa-check"></i><b>9.3</b> Tasks</a></li>
<li class="chapter" data-level="9.4" data-path="material.html"><a href="material.html#extra-material-on-miaverse-and-r-programming"><i class="fa fa-check"></i><b>9.4</b> Extra material on miaverse and R programming</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Microbiome data science with R/Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="importing-microbiome-data" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">5</span> Importing microbiome data<a href="importing-microbiome-data.html#importing-microbiome-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This section demonstrates how to import taxonomic profiling data in R.</p>
<div id="data-access" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Data access<a href="importing-microbiome-data.html#data-access" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><em>ADHD-associated changes in gut microbiota and brain in a mouse model</em></p>
<p>Tengeler AC <em>et
al.</em> (2020) <a href="https://doi.org/10.1186/s40168-020-00816-x"><strong>Gut microbiota from persons with
attention-deficit/hyperactivity disorder affects the brain in
mice</strong></a>. Microbiome
8:44.</p>
<p>In this study, mice are colonized with microbiota from participants
with ADHD (attention deficit hyperactivity disorder) and healthy
participants. The aim of the study was to assess whether the mice
display ADHD behaviors after being inoculated with ADHD microbiota,
suggesting a role of the microbiome in ADHD pathology.</p>
<p>Download the data from
<a href="https://github.com/microbiome/course_2022_radboud/tree/main/data">data</a>
subfolder.</p>
</div>
<div id="importing-microbiome-data-in-r" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Importing microbiome data in R<a href="importing-microbiome-data.html#importing-microbiome-data-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Import example data</strong> by modifying the examples in the online book
section on <a href="https://microbiome.github.io/OMA/data-introduction.html#loading-experimental-microbiome-data">data exploration and
manipulation</a>.</p>
<p>The data files in our example are in <em>biom</em> container, which is a
standard file format for microbiome data. Other file formats exist as
well, and import details vary by platform. Here, we import <em>biom</em> data
files into a specific data container (structure) in R,
<em>TreeSummarizedExperiment</em> (TreeSE) <a href="https://f1000research.com/articles/9-1246">Huang et
al. (2020)</a>.</p>
<p>The data container provides the basis for downstream data analysis in
the <em>miaverse</em> data science framework.</p>
<p><img src="https://raw.githubusercontent.com/FelixErnst/TreeSummarizedExperiment
/2293440c6e70ae4d6e978b6fdf2c42fdea7fb36a/vignettes/tse2.png" width="100%"/></p>
<p><strong>Figure sources:</strong></p>
<p><strong>Original article</strong>
- Huang R <em>et al</em>. (2021) <a href="https://doi.org/10.12688/%20f1000research.26669.2">TreeSummarizedExperiment: a S4 class
for data with hierarchical structure</a>. F1000Research 9:1246.</p>
<p><strong>Reference Sequence slot extension</strong>
- Lahti L <em>et al</em>. (2020) <a href="https://doi.org/10.7490/%20f1000research.1118447.1">Upgrading the R/Bioconductor ecosystem for microbiome
research</a> F1000Research 9:1464 (slides).</p>
<div id="example-solution" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Example solution<a href="importing-microbiome-data.html#example-solution" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let us first import the biom file into R / TreeSE container.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="importing-microbiome-data.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the mia R package</span></span>
<span id="cb4-2"><a href="importing-microbiome-data.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb4-3"><a href="importing-microbiome-data.html#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="importing-microbiome-data.html#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Defining file paths</span></span>
<span id="cb4-5"><a href="importing-microbiome-data.html#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Biom file (taxonomic profiles)</span></span>
<span id="cb4-6"><a href="importing-microbiome-data.html#cb4-6" aria-hidden="true" tabindex="-1"></a>biom_file_path <span class="ot">&lt;-</span> <span class="st">&quot;data/Tengeler2020/Aggregated_humanization2.biom&quot;</span></span>
<span id="cb4-7"><a href="importing-microbiome-data.html#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="importing-microbiome-data.html#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Import the data into SummarizedExperiment container</span></span>
<span id="cb4-9"><a href="importing-microbiome-data.html#cb4-9" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">loadFromBiom</span>(biom_file_path)</span>
<span id="cb4-10"><a href="importing-microbiome-data.html#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="importing-microbiome-data.html#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert this data to TreeSE container (no direct importer exists)</span></span>
<span id="cb4-12"><a href="importing-microbiome-data.html#cb4-12" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">as</span>(se, <span class="st">&quot;TreeSummarizedExperiment&quot;</span>)</span></code></pre></div>
<p>Check and clean up rowData (information on the taxonomic features).</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="importing-microbiome-data.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Investigate the rowData of this data object</span></span>
<span id="cb5-2"><a href="importing-microbiome-data.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(<span class="fu">rowData</span>(tse)))</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 6 columns
##             taxonomy1          taxonomy2           taxonomy3
##           &lt;character&gt;        &lt;character&gt;         &lt;character&gt;
## 1726470  &quot;k__Bacteria   p__Bacteroidetes      c__Bacteroidia
## 1726471  &quot;k__Bacteria   p__Bacteroidetes      c__Bacteroidia
## 17264731 &quot;k__Bacteria   p__Bacteroidetes      c__Bacteroidia
## 17264726 &quot;k__Bacteria   p__Bacteroidetes      c__Bacteroidia
## 1726472  &quot;k__Bacteria p__Verrucomicrobia c__Verrucomicrobiae
## 17264724 &quot;k__Bacteria   p__Bacteroidetes      c__Bacteroidia
##                      taxonomy4              taxonomy5           taxonomy6
##                    &lt;character&gt;            &lt;character&gt;         &lt;character&gt;
## 1726470       o__Bacteroidales      f__Bacteroidaceae     g__Bacteroides&quot;
## 1726471       o__Bacteroidales      f__Bacteroidaceae     g__Bacteroides&quot;
## 17264731      o__Bacteroidales  f__Porphyromonadaceae g__Parabacteroides&quot;
## 17264726      o__Bacteroidales      f__Bacteroidaceae     g__Bacteroides&quot;
## 1726472  o__Verrucomicrobiales f__Verrucomicrobiaceae     g__Akkermansia&quot;
## 17264724      o__Bacteroidales      f__Bacteroidaceae     g__Bacteroides&quot;</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="importing-microbiome-data.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># We notice that the rowData fields do not have descriptibve names.</span></span>
<span id="cb7-2"><a href="importing-microbiome-data.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Hence, let us rename the columns in rowData</span></span>
<span id="cb7-3"><a href="importing-microbiome-data.html#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">rowData</span>(tse)) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>, </span>
<span id="cb7-4"><a href="importing-microbiome-data.html#cb7-4" aria-hidden="true" tabindex="-1"></a>                        <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb7-5"><a href="importing-microbiome-data.html#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="importing-microbiome-data.html#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co"># We also notice that the taxa names are of form &quot;c__Bacteroidia&quot; etc.</span></span>
<span id="cb7-7"><a href="importing-microbiome-data.html#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Goes through the whole DataFrame. Removes &#39;.*[kpcofg]__&#39; from strings, where [kpcofg] </span></span>
<span id="cb7-8"><a href="importing-microbiome-data.html#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co"># is any character from listed ones, and .* any character.</span></span>
<span id="cb7-9"><a href="importing-microbiome-data.html#cb7-9" aria-hidden="true" tabindex="-1"></a>rowdata_modified <span class="ot">&lt;-</span> BiocParallel<span class="sc">::</span><span class="fu">bplapply</span>(<span class="fu">rowData</span>(tse), </span>
<span id="cb7-10"><a href="importing-microbiome-data.html#cb7-10" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">FUN =</span> stringr<span class="sc">::</span>str_remove, </span>
<span id="cb7-11"><a href="importing-microbiome-data.html#cb7-11" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">pattern =</span> <span class="st">&#39;.*[kpcofg]__&#39;</span>)</span>
<span id="cb7-12"><a href="importing-microbiome-data.html#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="importing-microbiome-data.html#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Genus level has additional &#39;\&quot;&#39;, so let&#39;s delete that also</span></span>
<span id="cb7-14"><a href="importing-microbiome-data.html#cb7-14" aria-hidden="true" tabindex="-1"></a>rowdata_modified <span class="ot">&lt;-</span> BiocParallel<span class="sc">::</span><span class="fu">bplapply</span>(rowdata_modified, </span>
<span id="cb7-15"><a href="importing-microbiome-data.html#cb7-15" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">FUN =</span> stringr<span class="sc">::</span>str_remove, </span>
<span id="cb7-16"><a href="importing-microbiome-data.html#cb7-16" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\&quot;</span><span class="st">&#39;</span>)</span>
<span id="cb7-17"><a href="importing-microbiome-data.html#cb7-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-18"><a href="importing-microbiome-data.html#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co"># rowdata_modified is a list, so convert this back to DataFrame format. </span></span>
<span id="cb7-19"><a href="importing-microbiome-data.html#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="co"># and assign the cleaned data back to the TSE rowData</span></span>
<span id="cb7-20"><a href="importing-microbiome-data.html#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(tse) <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(rowdata_modified)</span>
<span id="cb7-21"><a href="importing-microbiome-data.html#cb7-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-22"><a href="importing-microbiome-data.html#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Recheck rowData after the modifications</span></span>
<span id="cb7-23"><a href="importing-microbiome-data.html#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(<span class="fu">rowData</span>(tse)))</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 6 columns
##              Kingdom          Phylum            Class              Order
##          &lt;character&gt;     &lt;character&gt;      &lt;character&gt;        &lt;character&gt;
## 1726470     Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales
## 1726471     Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales
## 17264731    Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales
## 17264726    Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales
## 1726472     Bacteria Verrucomicrobia Verrucomicrobiae Verrucomicrobiales
## 17264724    Bacteria   Bacteroidetes      Bacteroidia      Bacteroidales
##                       Family           Genus
##                  &lt;character&gt;     &lt;character&gt;
## 1726470       Bacteroidaceae     Bacteroides
## 1726471       Bacteroidaceae     Bacteroides
## 17264731  Porphyromonadaceae Parabacteroides
## 17264726      Bacteroidaceae     Bacteroides
## 1726472  Verrucomicrobiaceae     Akkermansia
## 17264724      Bacteroidaceae     Bacteroides</code></pre>
<p>Next let us add sample information (colData) to our TreeSE object.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="importing-microbiome-data.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Sample phenodata</span></span>
<span id="cb9-2"><a href="importing-microbiome-data.html#cb9-2" aria-hidden="true" tabindex="-1"></a>sample_meta_file_path <span class="ot">&lt;-</span> <span class="st">&quot;data/Tengeler2020/Mapping_file_ADHD_aggregated.csv&quot;</span></span>
<span id="cb9-3"><a href="importing-microbiome-data.html#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="importing-microbiome-data.html#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check what type of data it is</span></span>
<span id="cb9-5"><a href="importing-microbiome-data.html#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co"># read.table(sample_meta_file_path)</span></span>
<span id="cb9-6"><a href="importing-microbiome-data.html#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="importing-microbiome-data.html#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co"># It seems like a comma separated file and it does not include headers</span></span>
<span id="cb9-8"><a href="importing-microbiome-data.html#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Let us read the file; note that sample names are in the first column</span></span>
<span id="cb9-9"><a href="importing-microbiome-data.html#cb9-9" aria-hidden="true" tabindex="-1"></a>sample_meta <span class="ot">&lt;-</span> <span class="fu">read.table</span>(sample_meta_file_path, <span class="at">sep=</span><span class="st">&quot;,&quot;</span>, <span class="at">header=</span><span class="cn">FALSE</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb9-10"><a href="importing-microbiome-data.html#cb9-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-11"><a href="importing-microbiome-data.html#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the data</span></span>
<span id="cb9-12"><a href="importing-microbiome-data.html#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(sample_meta))</span></code></pre></div>
<pre><code>##        V2       V3            V4   V5
## A110 ADHD Cohort_1 ADHD_Cohort_1 A110
## A12  ADHD Cohort_1 ADHD_Cohort_1  A12
## A15  ADHD Cohort_1 ADHD_Cohort_1  A15
## A19  ADHD Cohort_1 ADHD_Cohort_1  A19
## A21  ADHD Cohort_2 ADHD_Cohort_2  A21
## A23  ADHD Cohort_2 ADHD_Cohort_2  A23</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="importing-microbiome-data.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add headers for the columns (as they seem to be missing)</span></span>
<span id="cb11-2"><a href="importing-microbiome-data.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sample_meta) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;patient_status&quot;</span>, <span class="st">&quot;cohort&quot;</span>,</span>
<span id="cb11-3"><a href="importing-microbiome-data.html#cb11-3" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;patient_status_vs_cohort&quot;</span>, <span class="st">&quot;sample_name&quot;</span>)</span>
<span id="cb11-4"><a href="importing-microbiome-data.html#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="importing-microbiome-data.html#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Add this sample data to colData of the taxonomic data object</span></span>
<span id="cb11-6"><a href="importing-microbiome-data.html#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Note that the data must be given in a DataFrame format (required for our purposes)</span></span>
<span id="cb11-7"><a href="importing-microbiome-data.html#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse) <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(sample_meta)</span>
<span id="cb11-8"><a href="importing-microbiome-data.html#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="importing-microbiome-data.html#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the colData after modifications</span></span>
<span id="cb11-10"><a href="importing-microbiome-data.html#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(<span class="fu">colData</span>(tse)))</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 4 columns
##      patient_status      cohort patient_status_vs_cohort sample_name
##         &lt;character&gt; &lt;character&gt;              &lt;character&gt; &lt;character&gt;
## A110           ADHD    Cohort_1            ADHD_Cohort_1        A110
## A12            ADHD    Cohort_1            ADHD_Cohort_1         A12
## A15            ADHD    Cohort_1            ADHD_Cohort_1         A15
## A19            ADHD    Cohort_1            ADHD_Cohort_1         A19
## A21            ADHD    Cohort_2            ADHD_Cohort_2         A21
## A23            ADHD    Cohort_2            ADHD_Cohort_2         A23</code></pre>
<p>Add phylogenetic tree (rowTree).</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="importing-microbiome-data.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Phylogenetic tree</span></span>
<span id="cb13-2"><a href="importing-microbiome-data.html#cb13-2" aria-hidden="true" tabindex="-1"></a>tree_file_path <span class="ot">&lt;-</span> <span class="st">&quot;data/Tengeler2020/Data_humanization_phylo_aggregation.tre&quot;</span></span>
<span id="cb13-3"><a href="importing-microbiome-data.html#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="importing-microbiome-data.html#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the tree file</span></span>
<span id="cb13-5"><a href="importing-microbiome-data.html#cb13-5" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> ape<span class="sc">::</span><span class="fu">read.tree</span>(tree_file_path)</span>
<span id="cb13-6"><a href="importing-microbiome-data.html#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="importing-microbiome-data.html#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Add tree to rowTree</span></span>
<span id="cb13-8"><a href="importing-microbiome-data.html#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rowTree</span>(tse) <span class="ot">&lt;-</span> tree</span></code></pre></div>
<p>We can save the data object as follows.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="importing-microbiome-data.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tse, <span class="at">file=</span><span class="st">&quot;tse.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="loading-readily-processed-data" class="section level3 hasAnchor" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Loading readily processed data<a href="importing-microbiome-data.html#loading-readily-processed-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Alternatively, you can just load the already processed data set.</p>
<p>By using this readily processed data set we can skip the data import
step, and assume that the data has already been appropriately
preprocessed and available in the TreeSE container.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="importing-microbiome-data.html#cb15-1" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;data/Tengeler2020/tse.rds&quot;</span>)</span></code></pre></div>
<p>In addition to our example data, further demonstration data sets are
available in the TreeSE data container: see <a href="https://microbiome.github.io/OMA/containers.html#example-data">OMA demo data
sets</a>.</p>

<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: MatrixGenerics</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
##     colWeightedMeans, colWeightedMedians, colWeightedSds,
##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
##     rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     expand.grid, I, unname</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## 
## Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:MatrixGenerics&#39;:
## 
##     rowMedians</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## Loading required package: SingleCellExperiment</code></pre>
<pre><code>## Loading required package: TreeSummarizedExperiment</code></pre>
<pre><code>## Loading required package: Biostrings</code></pre>
<pre><code>## Loading required package: XVector</code></pre>
<pre><code>## 
## Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     strsplit</code></pre>
<pre><code>## Loading required package: MultiAssayExperiment</code></pre>
<pre><code>## Loading required package: ggplot2</code></pre>
<pre><code>## Loading required package: ggraph</code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:Biostrings&#39;:
## 
##     collapse, intersect, setdiff, setequal, union</code></pre>
<pre><code>## The following object is masked from &#39;package:XVector&#39;:
## 
##     slice</code></pre>
<pre><code>## The following object is masked from &#39;package:Biobase&#39;:
## 
##     combine</code></pre>
<pre><code>## The following objects are masked from &#39;package:GenomicRanges&#39;:
## 
##     intersect, setdiff, union</code></pre>
<pre><code>## The following object is masked from &#39;package:GenomeInfoDb&#39;:
## 
##     intersect</code></pre>
<pre><code>## The following objects are masked from &#39;package:IRanges&#39;:
## 
##     collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:S4Vectors&#39;:
## 
##     first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:BiocGenerics&#39;:
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following object is masked from &#39;package:matrixStats&#39;:
## 
##     count</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="reproducible-reporting-with-rmarkdown.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="microbiome-data-exploration.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["radboud2022_material.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
